plasma_framework/contracts/src/exits/utils/ExitId.sol

```diff
@@ -35,35 +36,24 @@ library ExitId {
     )
         internal
         pure
-        returns (uint160)
+        returns (uint168)
     {
+        bytes32 outputId;
         if (_isDeposit) {
-            bytes32 hashData = keccak256(abi.encodePacked(_txBytes, _utxoPos.encode()));
-            return _computeStandardExitId(hashData, _utxoPos.outputIndex);
+            outputId = OutputId.computeDepositOutputId(_txBytes, _utxoPos.outputIndex, _utxoPos.encode());
+        } else {
+            outputId = OutputId.computeNormalOutputId(_txBytes, _utxoPos.outputIndex);
         }
 
-        return _computeStandardExitId(keccak256(_txBytes), _utxoPos.outputIndex);
+        return uint168((uint256(outputId) >> (256 - FIRST_BIT)));
     }
 
     /**
     * @notice Given transaction bytes, returns in-flight exit ID
     * @param _txBytes Transaction bytes
     * @return Unique in-flight exit ID
     */
-    function getInFlightExitId(bytes memory _txBytes) internal pure returns (uint160) {
-        return uint160((uint256(keccak256(_txBytes)) >> 105).setBit(151));
-    }
-
-    function _computeStandardExitId(bytes32 _txhash, uint16 _outputIndex)
-        private
-        pure
-        returns (uint160)
-    {
-        uint256 exitId = (uint256(_txhash) >> 105) | (uint256(_outputIndex) << 152);
-        uint160 croppedExitId = uint160(exitId);
-
-        require(uint256(croppedExitId) == exitId, "ExitId overflows");
-
-        return croppedExitId;
+    function getInFlightExitId(bytes memory _txBytes) internal pure returns (uint168) {
+        return uint168((uint256(keccak256(_txBytes)) >> (256 - FIRST_BIT)).setBit(FIRST_BIT));
```

Comment:
same, first cast then shift?

Reply:
This is the way we've always done it to get the first 167 bits (previously 159 bits). See comment above